### PR TITLE
refactor: use pydantic model for job class

### PR DESCRIPTION
**Pull request by collinlokken**
## What does this pull request change?
Turn job class into pydantic model

## Why is this pull request needed?
Cleans up the code by removing the job.enity. The job object itself is a (de)serialized version of the dmss job entity.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,7 +16,9 @@ uvicorn = {extras = ["standard"], version = "^0.18.2"}
 azure-mgmt-containerinstance = "^10.0.0"
 azure-mgmt-resource = "^21.2.0"
 azure-identity = "^1.11.0"
-dm-cli = "^1.0.10"
+dm-cli = "^1.2.0"
+pydantic = "^1.10.12"
+mypy = "^1.7.0"
```
Comment (soofstad) [inline pyproject.toml]: Should not be here. Only a requirement for pre-commit

File: src/services/job_service.py [hunk 2]
```diff
@@ -73,29 +73,22 @@ def schedule_cron_job(job_scheduler: BackgroundScheduler, function: Callable, jo
...
 def _set_job(job: Job):
-    return get_job_store().set(str(job.job_uid), json.dumps(job.to_dict()))
-
-
-def load_cron_jobs():
-    for key in get_job_store().scan_iter():
-        job = _get_job(UUID(key.decode()))
-        if job.cron_job:
-            schedule_cron_job(scheduler, _run_job, job)
-            logger.info(f"Loaded and registered job '{job.job_uid}' from {config.SCHEDULER_REDIS_HOST}")
```
Comment (soofstad) [inline src/services/job_service.py]: This should still be in use. 
We need a way to re-register cron-jobs after restarting/deploying a new container

File: src/services/job_service.py [hunk 3]
```diff
@@ -165,18 +153,21 @@ def _run_job(job_uid: UUID) -> str:
...
+        update_document(
+            job.dmss_id, job.json(by_alias=True, exclude_none=True, exclude=job.exclude_keys), job.token
+        )  # Update in DMSS with status etc.
 
         _set_job(job)
         return job.log  # type: ignore
 
 
 def register_job(dmss_id: str) -> Tuple[str, str]:
+    print("hello dmms id:", dmss_id)
```
Comment (soofstad) [inline src/services/job_service.py]:
```suggestion
```

File: src/services/job_service.py [hunk 4]
```diff
@@ -223,8 +204,8 @@ def register_job(dmss_id: str) -> Tuple[str, str]:
         result = str(job.job_uid), "Job successfully started"
 
     _set_job(job)
-    job.update_entity_attributes()
-    update_document(job.dmss_id, job_entity, token=job.token)
+    print(job_entity)
```
Comment (soofstad) [inline src/services/job_service.py]:
```suggestion
```

**Review by soofstad — APPROVED** [review]
Looks good!

Now just making the tests run :D

File: src/services/job_service.py (hunk 2 above)
Comment (collinlokken) [reply src/services/job_service.py]: ok. I noticed it was not ues anywhere so i assumed it was old :)

**Pull request merged**